_commands/basics/gcc.md

```diff
@@ -0,0 +1,97 @@
+---
+---
+
+gcc flags
+--
+
+`gcc` is a compiler command that has the ability to execute options or 'Flags' when compiling.  These flags can be grouped and executed via the command line:
+
+`-o` places output in a file.
+
+`-g` turns on debugging information. 
+
+`-Werror` makes all warnings into errors.
+
+`-Wall` enables all the warnings.
+
+`-ar` creates an archive instead of a program.
+
+~~~ bash
+$ gcc test.c -o test
+$ gcc -g test.c
+$ gcc -Werror test.c
+$ gcc -Wall test.c
+$ gcc -ar test.c
+~~~
```

Comment:
Yes, great call.  I ended up switching the `-ar` flag to the `-O` flag.  The `-ar` was problematic. I thought the optimization was more straightforward and useful.  So, I combined the statements, like you commented.